htdocs/core/lib/files.lib.php

```diff
@@ -2747,12 +2747,9 @@ function dol_check_secure_access_document($modulepart, $original_file, $entity,
 	if (empty($modulepart)) {
 		return 'ErrorBadParameter';
 	}
+	// entity can't be empty
 	if (empty($entity)) {
-		if (!isModEnabled('multicompany')) {
-			$entity = 1;
-		} else {
-			$entity = 0;
-		}
+		$entity = $conf->entity;
```

Comment:
This looks to introduce bugs.
When modulepart is "medias", by definition it means it is the "public" directory (access by public with no login, no session and public for everybody). So there is only one "medias" directory (same for all entities). So for this special value of modulepart=medias, $entity must always be 1 and not $conf->entity.

Also when multicompany is on and a document is shared between entity 2 and 3 (for example a proposal), the PDF was createdand stored into entity 2. When we are in entity 3, we see the document and the PDF of entity 2 (because shared), but when we click on the link to PDF, it will try to download the file from the directory of entity 3 instead of 2. Did i miss something ?

Reply:
@eldy ok, the problem was elsewhere, look my last fix, thanks